packages/shell-api/src/shard.ts

```diff
@@ -0,0 +1,50 @@
+import Mongo from './mongo';
+import {
+  shellApiClassDefault,
+  hasAsyncChild,
+  ShellApiClass,
+} from './decorators';
+
+import {
+  Document
+} from '@mongosh/service-provider-core';
+import { MongoshUnimplementedError } from '@mongosh/errors';
+
+@shellApiClassDefault
+@hasAsyncChild
```

Comment:
this can be easily inferred from the other signatures, seems a detail we could hide in `@shellApiClassDefault`

Reply:
The `hasAsyncChild` attribute is just a helper to speed up inferring if there's a child attribute with `returnsPromise` (otherwise it could get quite slow checking every child of the shell api types, every time anything is passed through the writer). We could also infer it in the decorator, but I was thinking we could do a more major refactor where we use the typescript annotations to generate the attributes.

The `returnsPromise` and `returnType` decorators are pretty much always the same thing as the type signature. We could try to infer them dynamically, but I was thinking that'd be a much bigger refactor and would slow this down significantly so it should be done separately.